src/xo-app/vm/tab-disks.js

```diff
@@ -45,6 +43,72 @@ import {
   subscribeResourceSets
 } from 'xo'
 
+const COLUMNS_VM_PV = [
+  {
+    itemRenderer: vdi => <Text value={vdi.name_label} onChange={value => editVdi(vdi, { name_label: value })} />,
+    name: _('vdiNameLabel'),
+    sortCriteria: _ => _.name_label
+
+  },
+  {
+    itemRenderer: vdi => <Text value={vdi.name_description} onChange={value => editVdi(vdi, { name_description: value })} />,
+    name: _('vdiNameDescription'),
+    sortCriteria: _ => _.name_description
+
+  },
+  {
+    itemRenderer: vdi => <Size value={vdi.size || null} onChange={size => editVdi(vdi, { size })} />,
+    name: _('vdiSize'),
+    sortCriteria: _ => _.size
+
+  },
+  {
+    itemRenderer: (vdi, userData) => {
+      const vbds = userData[2].filter(vbd => vbd.id === vdi.$VBDs[0])
+      const vbd = vbds[0]
+      const sr = userData[0][vdi.$SR]
+      return sr &&
+      <XoSelect onChange={sr => migrateVdi(vdi, sr)}
+        xoType='SR'
+        predicate={sr => sr.$pool === vbd.$pool && isSrWritable(sr)}
+        labelProp='name_label'
+        value={sr}
+        useLongClick>
+        <Link to={`/srs/${sr.id}`}>{sr.name_label}</Link>
+      </XoSelect>
+    },
+    name: _('vdiSr')
+  },
+  {
+    itemRenderer: (vdi, userData) => {
+      const vbds = userData[2].filter(vbd => vbd.id === vdi.$VBDs[0])
+      const vbd = vbds[0]
+      return <Toggle value={vbd.bootable}
+        onChange={bootable => setBootableVbd(vbd, bootable)}
+          />
+    },
+    name: _('vbdBootableStatus')
+  },
+  {
+    itemRenderer: (vdi, userData) => {
+      const vbds = userData[2].filter(vbd => vbd.id === vdi.$VBDs[0])
+      const vbd = vbds[0]
+      return <StateButton
+        disabledLabel={_('vbdStatusDisconnected')}
+        disabledHandler={connectVbd}
+        disabledTooltip={_('vbdConnect')}
+        enabledLabel={_('vbdStatusConnected')}
+        enabledHandler={disconnectVbd}
+        enabledTooltip={_('vbdDisconnect')}
+        disabled={!(vbd.attached || isVmRunning(userData[1]))}
+        handlerParam={vbd}
+        state={vbd.attached} />
+    },
+    name: _('vbdStatus')
+  }
+]
+const COLUMNS = COLUMNS_VM_PV.filter(col => col.name.props.id !== 'vbdBootableStatus')
```

Comment:
`col.name` is an opaque (and changing) structure, you should not accessing its props.

If you want to identify a column, simply add an `id` field.